src/extensions/CanvasWidget/CanvasWidgetLayer.ts

```diff
@@ -260,29 +268,65 @@ export class CanvasWidgetLayer<LayerProps extends BaseLayerProps, State extends
     }
 }
 
-export const useCanvasWidgetLayer = <LayerProps extends BaseLayerProps, LayerState extends Object>(createLayer: () => CanvasWidgetLayer<LayerProps, LayerState>): CanvasWidgetLayer<LayerProps, LayerState> | null => {
+export const useLayer = <LayerProps extends BaseLayerProps, LayerState extends Object>(createLayer: () => CanvasWidgetLayer<LayerProps, LayerState>, layerProps?: LayerProps): CanvasWidgetLayer<LayerProps, LayerState> | null => {
```

Comment:
The layers now get their props in the call to useLayer. Now useLayer is a hook that will return a persistent layer object, but the layer props are re-set with every new render call. However, the onPropsChange is only called on the layer when the props actually change (according to a shallow check for modifications)